studio/components/interfaces/Organization/BillingSettings/BillingEmail.tsx

```diff
@@ -21,94 +22,97 @@ const BillingEmail = () => {
   const { name, billing_email } = selectedOrganization ?? {}
 
   const formId = 'org-billing-email'
-  const initialValues = {
-    billing_email: billing_email ?? '',
-  }
+  const initialValues = { billing_email: billing_email ?? '' }
 
   const canUpdateOrganization = useCheckPermissions(PermissionAction.UPDATE, 'organizations')
   const canReadBillingEmail = useCheckPermissions(PermissionAction.READ, 'organizations')
-  const { mutate: updateOrganization, isLoading: isUpdating } = useOrganizationUpdateMutation()
+  const { mutateAsync: updateOrganization, isLoading: isUpdating } = useOrganizationUpdateMutation()
 
-  const onUpdateOrganization = async (values: any, { setSubmitting, resetForm }: any) => {
+  const onUpdateOrganizationEmail = async (values: any, { resetForm }: any) => {
     if (!canUpdateOrganization) {
       return ui.setNotification({
         category: 'error',
         message: 'You do not have the required permissions to update this organization',
       })
     }
-
     if (!slug) return console.error('Slug is required')
+    if (!name) return console.error('Organization name is required')
 
-    updateOrganization(
-      { slug, billing_email: values.billing_email },
-      {
-        onSuccess: ({ billing_email }) => {
-          resetForm({ values: { billing_email }, initialValues: { billing_email } })
-          invalidateOrganizationsQuery(queryClient)
-          ui.setNotification({ category: 'success', message: 'Successfully saved settings' })
-        },
-      }
-    )
+    try {
+      const { billing_email } = await updateOrganization({
```

Comment:
nit: could we stick to the previous way of using `mutate` instead of `mutateAsync`
then we don't have the awkward try finally here

Reply:
I haven't actually touched this component - it only shows up as diff as I removed the old BillingSettings.tsx and renamed the BillingSettingsV2 to BillingSettings - GH wrongfully shows it as diff between these files, while it was really just a removal.

Let's do these changes separately